### PR TITLE
health: make Opsgenie API URL configurable

### PR DESCRIPTION
**Pull request by tinyhammers**
Add OPSGENIE_API_URL="" to so user can specify API endpoint

##### Summary

Fixes: #10556

##### Test Plan

This has been tested on our production infrastructure and is working

### CODE REVIEW

**Review by thiagoftsm — DISMISSED** [review]

File: health/notifications/health_alarm_notify.conf
Comment (Ferroin) [inline health/notifications/health_alarm_notify.conf:287]: This should probably be explicitly set to a sane default if possible.

Comment (thiagoftsm) [reply health/notifications/health_alarm_notify.conf:287]: I arrived in the same conclusion when I saw the PR #10562.

@tinyhammers , please ,add the previous URL as default value here.

Comment (tinyhammers) [reply health/notifications/health_alarm_notify.conf]: Sensible. Set to https://api.opsgenie.com

**Review by Ferroin — APPROVED** [review]
Approving in context, but I largely agree with @ilyam8 that this should probably be merged with #10562.

**Review by thiagoftsm — APPROVED** [review]

File: health/notifications/health_alarm_notify.conf
Comment (ilyam8) [inline health/notifications/health_alarm_notify.conf:287]: Please include all the changes from #10562 into this PR.

Don't forget to update [opsgenie documentation](https://github.com/netdata/netdata/tree/master/health/notifications/opsgenie#send-notifications-to-opsgenie) as well.

**Review by ilyam8 — CHANGES_REQUESTED** [review]

**Review by Ferroin — DISMISSED** [review]

File: health/notifications/health_alarm_notify.conf
Comment (tinyhammers) [reply health/notifications/health_alarm_notify.conf]: I've removed this OK to leave unset now.
Updated documentation to reflect new variable.

**Review by tinyhammers — COMMENTED** [review]
OK should be good to go, even replaced the ) I deleted by mistake 😂

**Review by Ferroin — DISMISSED** [review]

**Comment by Ferroin** [comment]
> OK should be good to go, even replaced the ) I deleted by mistake

Thanks for the work on this. It looks good to go from my perspective, but as of right now is probably not going to be merged until after the v1.29 release (currently scheduled for tomorrow) as we’re in a code freeze right now in preparation for the release and the hardcoded URL is not exactly a critical bug.

**Comment by tinyhammers** [comment]
Yeah, no worries @Ferroin. 
I should've pulled my finger out last week to get it in v1.29.

**Review by joelhans — DISMISSED** [review]
Docs part of this LGTM!

**Comment by thiagoftsm** [comment]
Hello @tinyhammers ,

I tested your PR right now with the old configuration I had from OPSGENIE, with this old configuration I had only `OPSGENIE_API_KEY` set and I got this error.

```
curl: (3) URL using bad/illegal format or missing URL
2021-02-01 15:57:22: alarm-notify.sh: ERROR: failed to send opsgenie notification for: hades test.chart.test_alarm is WARNING, with HTTP error code 000.
```

Considering this, the changes you brought is not keeping compatibily with old configuration, please, can you take a look on this before we merge your PR?

File: health/notifications/alarm-notify.sh.in
Comment (ilyam8) [inline health/notifications/alarm-notify.sh.in:394]: this line should be after

https://github.com/tinyhammers/netdata/blob/db489de43336d8c2bd576ee24124fd4053ed7018/health/notifications/alarm-notify.sh.in#L396-L413

**Review by ilyam8 — CHANGES_REQUESTED** [review]

File: health/notifications/alarm-notify.sh.in
Comment (ilyam8) [reply health/notifications/alarm-notify.sh.in:394]: @thiagoftsm that should fix the issue you mentioned ^^

Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:394]: This is not happening @ilyam8 :

```bash
bash-5.1# grep "https://api.opsgenie.com" alarm-notify.sh 
OPSGENIE_API_URL=${OPSGENIE_API_URL:-"https://api.opsgenie.com"}
bash-5.1# ./alarm-notify.sh test

# SENDING TEST WARNING ALARM TO ROLE: sysadmin
2021-02-01 16:48:21: alarm-notify.sh: INFO: sent slack notification for: hades test.chart.test_alarm is WARNING to '#myalarms'
2021-02-01 16:48:22: alarm-notify.sh: INFO: sent telegram notification for: hades test.chart.test_alarm is WARNING to '-363762934'
curl: (3) URL using bad/illegal format or missing URL
2021-02-01 16:48:22: alarm-notify.sh: ERROR: failed to send opsgenie notification for: hades test.chart.test_alarm is WARNING, with HTTP error code 000.

```

Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:394]: I ran now another test with master branch:

```bash
bash-5.1# grep "https://api.opsgenie.com" alarm-notify.sh 
  httpcode=$(docurl -X POST -H "Content-Type: application/json" -d "${payload}" "https://api.opsgenie.com/v1/json/integrations/webhooks/netdata?apiKey=${OPSGENIE_API_KEY}")
bash-5.1# ./alarm-notify.sh test

# SENDING TEST WARNING ALARM TO ROLE: sysadmin
2021-02-01 17:05:39: alarm-notify.sh: INFO: sent slack notification for: hades test.chart.test_alarm is WARNING to '#myalarms'
2021-02-01 17:05:40: alarm-notify.sh: INFO: sent telegram notification for: hades test.chart.test_alarm is WARNING to '-99999'
2021-02-01 17:05:40: alarm-notify.sh: INFO: sent opsgenie notification for: hades test.chart.test_alarm is WARNING
# OK
```

Everything worked, so the PR needs to be fixed.

Comment (ilyam8) [reply health/notifications/alarm-notify.sh.in:394]: > This is not happening @ilyam8 :

I said that we need to put `OPSGENIE_API_URL=${OPSGENIE_API_URL:-"https://api.opsgenie.com"}` after `health_alarm_notify.conf` loading. Did you try it?

---

Take attention to the line numbers

```cmd
[ilyam@pc plugins.d]$ grep -n "https://api.opsgenie.com" alarm-notify.sh
394:#OPSGENIE_API_URL=${OPSGENIE_API_URL:-"https://api.opsgenie.com"}
415:OPSGENIE_API_URL=${OPSGENIE_API_URL:-"https://api.opsgenie.com"}
```

```cmd
[ilyam@pc plugins.d]$ ./alarm-test.sh

# SENDING TEST WARNING ALARM TO ROLE: sysadmin
2021-02-02 11:55:27: alarm-notify.sh: WARNING: Cannot find file '/opt/netdata/etc/netdata/health_alarm_notify.conf'.
2021-02-02 11:55:27: alarm-notify.sh: ERROR: failed to send opsgenie notification for: pc test.chart.test_alarm is WARNING, with HTTP error code 400.
# FAILED

# SENDING TEST CRITICAL ALARM TO ROLE: sysadmin
2021-02-02 11:55:27: alarm-notify.sh: WARNING: Cannot find file '/opt/netdata/etc/netdata/health_alarm_notify.conf'.
2021-02-02 11:55:28: alarm-notify.sh: ERROR: failed to send opsgenie notification for: pc test.chart.test_alarm is CRITICAL, with HTTP error code 400.
# FAILED

# SENDING TEST CLEAR ALARM TO ROLE: sysadmin
2021-02-02 11:55:28: alarm-notify.sh: WARNING: Cannot find file '/opt/netdata/etc/netdata/health_alarm_notify.conf'.
2021-02-02 11:55:28: alarm-notify.sh: ERROR: failed to send opsgenie notification for: pc test.chart.test_alarm is CLEAR, with HTTP error code 400.
# FAILED
```

Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:394]: @ilyam8 I tested now and when we move it works.

@tinyhammers , please, move the variable declaration for the line 415 a @ilyam8 described in the previous comment.

**Review by ilyam8 — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
After the last commit, I can verify that everything is working with old configuration.

Thank you very much @tinyhammers !

**Comment by Ferroin** [comment]
The merge of this will either have to wait until we’re out of the post-release soft code freeze, or would require changing the target branch from `master` to `develop` as it’s technically a new feature and not a bugfix.

@tinyhammers It’s up to you whether you want to wait the week or two for the soft freeze to end or just rebase on top of `develop`.

**Comment by tinyhammers** [comment]
Happy to wait :)

**Comment by OdysLam** [comment]
Now that we got the release out, we are good to go @Ferroin ?

**Comment by Ferroin** [comment]
> Now that we got the release out, we are good to go @Ferroin ?

We’re waiting until tomorrow (2020-02-11) to verify that no other major bugs are discovered after the recent patch release, but this PR is right at the top of my list to merge once the code freeze is over.

**Pull request merged**
